osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallenge.cs

```diff
@@ -405,6 +415,29 @@ private void trySetDailyChallengeBeatmap()
                 Schedule(forcefullyExit);
         });
 
+        private void dailyChallengeChanged(ValueChangedEvent<DailyChallengeInfo?> change)
+        {
+            if (change.OldValue?.RoomID == room.RoomID.Value && change.NewValue == null)
+            {
+                notificationOverlay?.Post(waitForNextChallengeNotification = new SimpleNotification
+                {
+                    Text = "Today's daily challenge has concluded. Thanks for playing! The next one should appear in a few minutes."
+                });
+            }
+
+            if (change.NewValue != null && change.NewValue.Value.RoomID != room.RoomID.Value)
+            {
+                var roomRequest = new GetRoomRequest(change.NewValue.Value.RoomID);
+
+                roomRequest.Success += room =>
+                {
+                    waitForNextChallengeNotification?.Close(false);
+                    notificationOverlay?.Post(new NewDailyChallengeNotification(room));
+                };
+                API.Queue(roomRequest);
```

Comment:
See last line of OP:

> The notifications are constrained to the daily challenge screen only to not spam users who may not care.

I'd rather avoid accusations of us 'railroading' users into this feature, but maybe I'm overly sensitive.

Reply:
fair i guess. we'll see if people request it :)

Reply:
Maybe only show for those who participated?

The current behavior doesn't work for people who log in straight at 0 UTC, when the button doesn't show.

Reply:
I'm honestly thinking the vast majority of users are not going to be annoyed by a notification once a day at a specific time of day and we should just go for it and wait for complaints, rather than the other way around.

Obviously only for the "new daily challenge" notification, not the other one.

Reply:
Ok well I'll have a go I guess, although now that I think about it figuring out a place to put the notification logic may prove to be a right pain in the ass.

Reply:
I've done this in https://github.com/ppy/osu/pull/29188/commits/1b57a2a136c626939669a1a6f6e425b93f823ebe. No toggle for now unless you think it is imperative for one to exist.

Not sure you'll like the placement, but it _does_ work globally because of some implementation details (even if the button is offscreen, the bindable change callbacks fire, because that stuff is being invoked from `MetadataClient` which is everpresent).